runtime/drivers/sqlite/sqlite.go

```diff
@@ -55,7 +56,21 @@ func (d driver) Drop(config map[string]any, logger *zap.Logger) error {
 }
 
 func (d driver) Spec() drivers.Spec {
-	return drivers.Spec{}
+	return drivers.Spec{
+		DisplayName: "SQLite",
+		Description: "Import data from SQLite table to DuckDB.",
+		SourceProperties: []drivers.PropertySchema{
+			{
+				Key:         "sql",
+				Type:        drivers.StringPropertyType,
+				Required:    true,
+				DisplayName: "SQL",
+				Description: "Query to extract data from SQLite",
+				Placeholder: "SELECT * FROM sqlite_scan('sqlite.db', 'film');",
```

Comment:
I just realised this is still problematic – it has the problem described in bullet 2 [here](https://www.notion.so/rilldata/Extend-our-existing-DuckDB-connector-to-support-Postgres-and-SQLite-343844b9024a4051bdc6f971ccdd4d59?pvs=4#261768be31ff4afeb5969df34093daf5).

If it has `connector: sqlite`, you would expect `sql: <sqlite query>`, and should at least have a path to generic support for other OLAPs.

If this is a DuckDB-only thing, then it should probably be part of the `duckdb` connector (could just check if the source contains `sqlite_scan` and opportunistically load the extension if it has.

Reply:
Good point.

> If this is a DuckDB-only thing, then it should probably be part of the duckdb connector (could just check if the source contains sqlite_scan and opportunistically load the extension if it has.

I think doing this also just fulfils half the criteria listed in the point. We would have a `sqlite` connector but one that only works with `duckdb`. For a user it will also be inconsistent to use connector type as `postgres` for `postgres` connector and `duckdb` for `sqlite` connector.
I would suggest we take approach similar to postgres then for sqlite as well.

Reply:
Sorry, what I meant by "be part of the duckdb connector" is that it would be in a source with `connector: duckdb`. So it would fulfill the criteria (I think), but not be a standalone connector.